packages/react-devtools-shared/src/__tests__/profilingCache-test.js

```diff
@@ -749,4 +754,671 @@ describe('ProfilingCache', () => {
+          "didPropsChange": false,
+          "didPropsDeepChange": false,
+          "didStateChange": false,
+          "didStateDeepChange": false,
+          "isFirstMount": true,
+          "nonMemoizedProps": null,
+        },
+        4 => Object {
+          "didContextChange": false,
+          "didContextDeepChange": false,
+          "didHooksChange": false,
+          "didPropsChange": false,
+          "didPropsDeepChange": false,
+          "didStateChange": false,
+          "didStateDeepChange": false,
+          "isFirstMount": true,
+          "nonMemoizedProps": null,
+        },
+        5 => Object {
+          "didContextChange": false,
+          "didContextDeepChange": false,
+          "didHooksChange": false,
+          "didPropsChange": false,
+          "didPropsDeepChange": false,
+          "didStateChange": false,
+          "didStateDeepChange": false,
+          "isFirstMount": true,
+          "nonMemoizedProps": null,
+        },
+        6 => Object {
+          "didContextChange": false,
+          "didContextDeepChange": false,
+          "didHooksChange": false,
+          "didPropsChange": false,
+          "didPropsDeepChange": false,
+          "didStateChange": false,
+          "didStateDeepChange": false,
+          "isFirstMount": true,
+          "nonMemoizedProps": null,
+        },
+      }
+    `);
+
+    // Second render. Only non memoized component updates didPropsChange is false.
+    expect(getPerfInsightSnapshot(1)).toMatchInlineSnapshot(`
+      Map {
+        2 => Object {
+          "didContextChange": false,
+          "didContextDeepChange": false,
+          "didHooksChange": false,
+          "didPropsChange": false,
+          "didPropsDeepChange": false,
+          "didStateChange": false,
+          "didStateDeepChange": false,
+          "isFirstMount": false,
+          "nonMemoizedProps": null,
+        },
+      }
+    `);
+
+    // Third render. We increment count and expect props to deep change.
+    expect(getPerfInsightSnapshot(2)).toMatchInlineSnapshot(`
+      Map {
+        2 => Object {
+          "didContextChange": false,
+          "didContextDeepChange": false,
+          "didHooksChange": false,
+          "didPropsChange": true,
+          "didPropsDeepChange": true,
+          "didStateChange": false,
+          "didStateDeepChange": false,
+          "isFirstMount": false,
+          "nonMemoizedProps": null,
+        },
+        3 => Object {
+          "didContextChange": false,
+          "didContextDeepChange": false,
+          "didHooksChange": false,
+          "didPropsChange": true,
+          "didPropsDeepChange": true,
+          "didStateChange": false,
+          "didStateDeepChange": false,
+          "isFirstMount": false,
+          "nonMemoizedProps": null,
+        },
+      }
+    `);
+
+    // Fourth render. We update style and except NeedsDeepMemo to update,
+    // telling us didPropsDeepChange to be false
```

Comment:
This also looks to be the opposite of what I'd expect. This _is_ a deep change, so why is `didPropsDeepChange: false`? (For what matter, why is `didPropsChange: false` also for one of the components below?)